### PR TITLE
JSUI-3266 Remove handshakeToken from url after exchange

### PR DESCRIPTION
**Pull request by samisayegh**
https://coveord.atlassian.net/browse/JSUI-3266

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: unitTests/ui/AuthenticationProviderTest.ts [hunk 1]
```diff
@@ -195,6 +195,7 @@ export function AuthenticationProviderTest() {
         const logggerSpy = spyOn(test.cmp.logger, 'error');
         triggerAfterComponentsInitialization();
 
+        await Promise.resolve();
         await Promise.resolve();
```
Comment (samisayegh) [inline unitTests/ui/AuthenticationProviderTest.ts:199]: Adding [this then chain](https://github.com/coveo/search-ui/pull/1765/files#diff-17ddd73583344050f04a376ac6631defe9aa8cc84e12ca323b1c4619488a2279R176) failed this test unless there was either a second `Promise.resolve` or a `setTimeout` 🤷

File: src/ui/AuthenticationProvider/AuthenticationProvider.ts [hunk 2]
```diff
@@ -188,6 +197,19 @@ export class AuthenticationProvider extends Component {
     localStorage.setItem(accessTokenStorageKey, accessToken);
   }
 
+  private removeHandshakeTokenFromUrl() {
```
Comment (olamothe) [inline src/ui/AuthenticationProvider/AuthenticationProvider.ts:200]: Could this be replaced by a regex ? 

Not sure this works for all cases, please run the UT first. I assume the token is always alpha numerical.

```javascript
const adjustedHash = hash.replace(/\&?handshake_token=[a-z0-9]+\&?/, '')

[...]
```


Or do you find that more complex overall ?

**Review by olamothe — APPROVED** [review]
Why was this change needed ?

Is it just for "URL cleanliness" aspect or were there more important reason ?

**Review by ThibodeauJF — APPROVED** [review]

File: src/ui/AuthenticationProvider/AuthenticationProvider.ts (hunk 2 above)
Comment (samisayegh) [reply src/ui/AuthenticationProvider/AuthenticationProvider.ts:200]: Hmm, a regex would be more elegant, but I'm not sure how to manage the case where the handshake token is in the middle of the hash:

```
#a=b&handshake_token=token&c=d
```
The regex would adjust the string to:
```
a=bc=d
```

We can assume the token is always at the end, but if the search api ever changes this, we might need to adjust the UI code

**Comment by samisayegh** [comment]
@olamothe, yep the Intuit team asked if we could clean up the url post-handshake. It would also prevent trying to handshake on page refresh,

File: unitTests/ui/AuthenticationProviderTest.ts (hunk 1 above)
Comment (btaillon) [reply unitTests/ui/AuthenticationProviderTest.ts:199]: you can use a `resolveAfter`

**Review by btaillon — APPROVED** [review]

File: unitTests/ui/AuthenticationProviderTest.ts (hunk 1 above)
Comment (samisayegh) [reply unitTests/ui/AuthenticationProviderTest.ts:199]: oh! I'll give that a try

Comment (samisayegh) [reply unitTests/ui/AuthenticationProviderTest.ts:199]: oh! I'll give that a try, thanks

**Pull request merged**